packages/strapi-connector-bookshelf/lib/buildQuery.js

```diff
@@ -14,15 +14,23 @@ const BOOLEAN_OPERATORS = ['or'];
  * @param {Object} options.filters - Filters params (start, limit, sort, where)
  */
 const buildQuery = ({ model, filters }) => qb => {
+  const joinsTree = buildJoinsAndFilter(qb, model, filters);
+
   if (_.has(filters, 'where') && Array.isArray(filters.where) && filters.where.length > 0) {
     qb.distinct();
   }
 
-  const joinsTree = buildJoinsAndFilter(qb, model, filters);
-
   if (_.has(filters, 'sort')) {
-    const clauses = filters.sort.map(buildSortClauseFromTree(joinsTree));
-    qb.orderBy(reject(isEmpty, clauses));
+    const clauses = filters.sort.map(buildSortClauseFromTree(joinsTree)).filter(c => !isEmpty(c));
+    const orderBy = clauses.map(({ order, alias }) => ({ order, column: alias }));
+    const orderColumns = clauses
+      .filter(({ alias, column }) => alias !== column) // avoid renaming columns of the main table
```

Comment:
I don't understand why you need this ? if the alias equal the column then you are not renaming anything anyway ?

Reply:
Not really, if the column is in the main table, than it would be named to "mainTableName.columnName" and would be a duplicated of the column. We would find this column in the query results (it breaks some tests without this line)